### PR TITLE
Upgraded project to new C# project format, added multiple-targets

### PR DESCRIPTION
**Pull request by JordanSolutions**

### CODE REVIEW

**Comment by JordanSolutions** [comment]
Forgot to mention, that also includes multi-framework targets, producing a very nice nuget package output with all targets included from 3.5 to netstandards 2.0

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,6 @@ packages/
 
 #ignore Resharper folders
 _ReSharper*/
+/.vs/Open.Nat/DesignTimeBuild
+/.vs/Open.Nat/v15/Server/sqlite3
+/Open.Nat/.vs/Open.Nat.old/v15/Server/sqlite3
```
Comment (lontivero) [inline .gitignore]: Could you revert this change, please? Otherwise, probably could be a good idea to use a pattern like `.vs`

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,6 @@ packages/
 
 #ignore Resharper folders
 _ReSharper*/
+/.vs/Open.Nat/DesignTimeBuild
```
Comment (lontivero) [inline .gitignore]: Could you revert this change, please? Otherwise, probably could be a good idea to use a pattern like `.vs`

Comment (lontivero) [inline .gitignore]: Could you revert this change, please? Otherwise, probably could be a good idea to use a pattern like `.vs`

Comment (lontivero) [inline .gitignore]: Could you revert this change, please? Otherwise, probably could be a good idea to use a pattern like `.vs`

Comment (lontivero) [inline .gitignore]: Could you remove these lines and replace them by something like `.vs`, please?

Comment (lontivero) [inline .gitignore]: Could you replace these lines by something like `.vs`, please?

File: Open.Nat/AssemblyInfoAdditions.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿using System.Runtime.CompilerServices;
```
Comment (lontivero) [inline Open.Nat/AssemblyInfoAdditions.cs]: Is this new file necessary? I see the same is specified in the Open.Nat.csproj file.

File: Open.Nat/Open.Nat.csproj [hunk 4]
```diff
@@ -1,160 +1,45 @@
...
-    <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
-    <WarningLevel>0</WarningLevel>
-    <DocumentationFile>bin\Release\net35\Open.Nat.XML</DocumentationFile>
-  </PropertyGroup>
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)'!='netstandard2.0'">
     <Reference Include="System" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Xml.Linq" />
+    <Reference Include="System.Data.DataSetExtensions" />
```
Comment (lontivero) [inline Open.Nat/Open.Nat.csproj]: What do we need `System.Data.DataSetExtensions` for?

File: Open.Nat/Open.Nat.csproj [hunk 5]
```diff
@@ -1,160 +1,45 @@
...
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
-	 <Reference Include="System.Threading, Version=1.0.2856.102, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL"
```
Comment (lontivero) [inline Open.Nat/Open.Nat.csproj]: .NET 3.5 needs TaskParallelLibrary. Did you compiled and run Open.NAT with NET 3.5?

Comment (lontivero) [reply Open.Nat/Open.Nat.csproj]: Forget it. I saw it bellow.

File: .gitignore (hunk 1 above)
Comment (JordanSolutions) [reply .gitignore]: Will do, sorry about that.

File: Open.Nat.sln [hunk 6]
```diff
@@ -34,14 +34,14 @@ Global
...
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Debug 3.5|Any CPU.ActiveCfg = Debug 3.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Debug 3.5|Any CPU.Build.0 = Debug 3.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Debug 4.5|Any CPU.ActiveCfg = Debug 4.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Debug 4.5|Any CPU.Build.0 = Debug 4.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Release 3.5|Any CPU.ActiveCfg = Release 3.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Release 3.5|Any CPU.Build.0 = Release 3.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Release 4.5|Any CPU.ActiveCfg = Release 4.5|Any CPU
-		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Release 4.5|Any CPU.Build.0 = Release 4.5|Any CPU
+		{F5D74163-145F-47BF-83DC-D0E07249C6CA}.Debug 3.5|Any CPU.ActiveCfg = Debug|Any CPU
```
[the remark this answers is not in the document]
Comment (JordanSolutions) [reply Open.Nat.sln]: it seams that VS shifted things around when upgraded from 14 to 15 since the new C# project format is not compatible (as far as I know) with VS2015 (14). All is still available just shifted it seams.

File: Open.Nat/AssemblyInfoAdditions.cs (hunk 3 above)
Comment (JordanSolutions) [reply Open.Nat/AssemblyInfoAdditions.cs]: This is true for the most part, but the caveat was your test cases use internal functions and/or classes, I had to add the file in order to add InternalsVisibleTo but that is the only statement in there. The rest of the metadata shifted to the csproj file.

File: Open.Nat/Open.Nat.csproj (hunk 4 above)
Comment (JordanSolutions) [reply Open.Nat/Open.Nat.csproj]: Removing, my bad, I was upgrading my own project at the same time, copy+past, didn't notice, same goes for System.Data
